### DISCUSSION TITLE
ESPNow ESP32 returning 'ESP_ERR_ESPNOW_ARG'

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#10594] [ESP32]
I am enhancing my ESPNow repeater project so that the remote and
repeater can go to deepsleep().  I plan to use DS3231s on the remote
and repeater so that I can wake them up each hour within +/- a few
seconds to obtain end-to-end comms.

During testing I can get the remote to run without errors when there
are no peers.  However, the repeater has a very long script where I
am doing multiple e0.recv() and e0.send() ... the calls to e0.recv()
will timeout (10 secs) when there is no host and 4 * e0.send() work
without error, but a 5th e0.send() gives the following error: 

`E (36810) ESPNOW: Invalid argument!
Error during execution: (-12390, 'ESP_ERR_ESPNOW_ARG')`

That 5th e0.send() is the same format as the others.

Is there some issue with the number of Falses that can be returned
by e0.send()?

Thanks

**Comment by davefes** [comment]
Further testing indicates that one is not allowed to put an empty string in e0.send(), ie
```py
remote_status = '' #  empty string

eo.send(remote_mac, remote_status, True)
```
which I was doing in stand-alone device testing.

**Reply by glenn20** [reply]
Yes - this is a constraint of the espressif ESPNOW api.

**Reply by davefes** [reply]
Thank you.
